### PR TITLE
docs: document the test suite

### PR DESCRIPTION
**Pull request by FFY00**
Please proofread, I have dyslexia so there may be some obvious typos I could not identify. Feel free to suggest how to improve the current draft, but don't block this PR if you want to propose any major change, we'll merge this and you can submit a followup PR.

### CODE REVIEW

File: docs/test_suite.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+`tox documentation`_.
+
+We have a fairly big environment matrix. We run tests for all supported Python
+versions and implementations, and with the module being invoked by either path,
+sdist install, and wheel install. Additionally, we have an environment for type
+checking, and one to produce the documentation. There are some other extra
+environments, like checking the code with the minimum version of each
+dependency.
+
+Some examples commands for this project:
```
Comment (henryiii) [inline docs/test_suite.rst:38]:
```suggestion
Some examples commands for this project:

```

Doesn't render as a list with no space.

File: docs/test_suite.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+.. code-block:: console
+
+     tox
+
+
+You can find out more about how to run ``tox`` and its arguments in the
+`tox documentation`_.
+
+We have a fairly big environment matrix. We run tests for all supported Python
+versions and implementations, and with the module being invoked by either path,
```
Comment (layday) [inline docs/test_suite.rst:32]:
```suggestion
versions and implementations, and with the module being invoked from path,
```

File: docs/test_suite.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
+**********
+Test Suite
+**********
+
+Due to its nature, ``build`` has a somewhat complex test suite, which we will
+try to go through in this document.
+
+Firstly, there are two set of tests, unit tests and integration tests. In unit
+tests, we test the actual code implementation. In integration tests, we test
+``build`` on a few real world projects, this is mostly a sanity test.
```
Comment (henryiii) [inline docs/test_suite.rst:10]:
```suggestion
``build`` on a few real world projects; this is mostly a sanity test.
```

File: docs/test_suite.rst [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+.. code-block:: console
+
+     tox
+
+
+You can find out more about how to run ``tox`` and its arguments in the
+`tox documentation`_.
+
+We have a fairly big environment matrix. We run tests for all supported Python
```
Comment (henryiii) [inline docs/test_suite.rst:31]:
```suggestion
We have a fairly large environment matrix. We run tests for all supported Python
```

File: docs/test_suite.rst [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+versions and implementations, and with the module being invoked by either path,
+sdist install, and wheel install. Additionally, we have an environment for type
```
Comment (henryiii) [inline docs/test_suite.rst:33]:
```suggestion
sdist install, or wheel install. Additionally, we have an environment for type
```

**Review by henryiii — COMMENTED** [review]
What about adding a tox list environments example?

**Review by gaborbernat — APPROVED** [review]

**Review by gaborbernat — APPROVED** [review]

**Review by gaborbernat — APPROVED** [review]

**Comment by FFY00** [comment]
> What about adding a tox list environments example?

Feel free to submit a PR :)

**Pull request merged**